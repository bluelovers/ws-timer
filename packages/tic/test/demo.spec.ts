import Tic from '../index';

test(`should run interval x 3 , timeout x 1`, () =>
{
	expect.assertions(4 + 2)

	const tic = new Tic();

	expect(tic).toMatchSnapshot();

	tic.interval((thing, key, param1) =>
	{
		expect(thing).toHaveProperty('limit', Infinity)
	}, 5000, 'Every');

	tic.add((thing, key, param1) =>
	{
		expect(thing).toHaveProperty('limit', 1)
	}, 5000);

	tic.tick(15000);

	expect(tic).toMatchSnapshot();

});

test(`auto tick`, () =>
{
	expect.assertions(4 + 2)

	const tic = new Tic();

	expect(tic).toMatchSnapshot();

	tic.interval((thing, key, param1) =>
	{
		expect(thing).toHaveProperty('limit', Infinity)
	}, 5000, 'Every');

	tic.add((thing, key, param1) =>
	{
		expect(thing).toHaveProperty('limit', 1)
	}, 5000);

	tic.tick(15000-4000);

	tic.tick();

	expect(tic).toMatchSnapshot();

});

