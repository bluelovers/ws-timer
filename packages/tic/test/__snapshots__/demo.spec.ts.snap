// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auto tick 1`] = `
Tic {
  "_dt": -1,
  "_id": 0,
  "_now": 0,
  "_things": Map {},
}
`;

exports[`auto tick 2`] = `
Tic {
  "_dt": 4000,
  "_id": 2,
  "_next": 20000,
  "_now": 15000,
  "_things": Map {
    1 => Object {
      "args": Array [
        "Every",
      ],
      "at": 20000,
      "elapsed": 0,
      "fn": [Function],
      "limit": Infinity,
      "timeout": 5000,
    },
  },
}
`;

exports[`should run interval x 3 , timeout x 1 1`] = `
Tic {
  "_dt": -1,
  "_id": 0,
  "_now": 0,
  "_things": Map {},
}
`;

exports[`should run interval x 3 , timeout x 1 2`] = `
Tic {
  "_dt": 15000,
  "_id": 2,
  "_next": 20000,
  "_now": 15000,
  "_things": Map {
    1 => Object {
      "args": Array [
        "Every",
      ],
      "at": 20000,
      "elapsed": 15000,
      "fn": [Function],
      "limit": Infinity,
      "timeout": 5000,
    },
  },
}
`;
